src/utils.js

```diff
@@ -58,6 +58,36 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  // Finds variable reference in current scope
+  for (let variable of variables) {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+      break;
+    }
+  }
+
+  if (scopeVar && scopeVar.defs && scopeVar.defs[0] &&
+      scopeVar.defs[0].node && scopeVar.defs[0].node.init &&
+      scopeVar.defs[0].node.init !== null) {
+    // Return the name of the first definition of the variable which
+    // corresponds to the node passed in.
```

Comment:
The first item in defs has always shown to be the first use of the variable (and therefore the definition). I suppose if we want to be safe I can check each item in `defs` and try and see which ones are variable assignments and return the last variable assignment used. (which would be the last in that list? docs aren't really clear but that's what it looks like to me).